homeassistant/components/deconz/sensor.py

```diff
@@ -245,38 +292,40 @@ def extra_state_attributes(self) -> dict[str, bool | float | int | None]:
         return attr
 
 
-class DeconzTemperature(DeconzDevice, SensorEntity):
-    """Representation of a deCONZ temperature sensor.
-
-    Extra temperature sensor on certain Xiaomi devices.
-    """
+class DeconzPropertySensor(DeconzDevice, SensorEntity):
+    """Representation of a deCONZ secondary attribute sensor."""
 
     TYPE = DOMAIN
     _device: PydeconzSensor
 
-    def __init__(self, device: PydeconzSensor, gateway: DeconzGateway) -> None:
-        """Initialize deCONZ temperature sensor."""
+    def __init__(
+        self,
+        device: PydeconzSensor,
+        gateway: DeconzGateway,
+        description: DeconzSensorDescription,
+    ) -> None:
+        """Initialize deCONZ sensor."""
+        self.entity_description = description
         super().__init__(device, gateway)
 
-        self.entity_description = ENTITY_DESCRIPTIONS[Temperature]
-        self._attr_name = f"{self._device.name} Temperature"
+        self._attr_name = f"{self._device.name} {description.suffix}"
+        self._update_keys = {description.update_key, "reachable"}
 
     @property
     def unique_id(self) -> str:
         """Return a unique identifier for this device."""
-        return f"{self.serial}-temperature"
+        return f"{self.serial}-{self.entity_description.suffix.lower()}"
 
     @callback
     def async_update_callback(self) -> None:
         """Update the sensor's state."""
-        keys = {"temperature", "reachable"}
-        if self._device.changed_keys.intersection(keys):
+        if self._device.changed_keys.intersection(self._update_keys):
             super().async_update_callback()
 
     @property
     def native_value(self) -> StateType:
         """Return the state of the sensor."""
-        return self._device.secondary_temperature  # type: ignore[no-any-return]
+        return getattr(self._device, self.entity_description.device_property)  # type: ignore[no-any-return]
```

Comment:
You could add a `value_fn` lambda to the entity description, removing the need to do all that `getattr` magic stuff.

Reply:
Sweet! You don't happen. To have an example at hand?

Reply:
search for `value_fn` 😄 

`tailscale` maybe?

Reply:
Easy enough :)